includes/class-wc-stripe-helper.php

```diff
@@ -1785,6 +1785,44 @@ public static function is_currency_supported_for_indian_recurring_payment_mandat
 		return in_array( strtolower( $currency ), $supported_currencies, true );
 	}
 
+	/**
+	 * Checks if the payment method should be saved.
+	 *
+	 * @since 9.6.0
+	 * @param bool $force_save Whether the payment method should be saved.
+	 * @param string $order_id Order ID.
+	 * @return bool
+	 */
+	public static function should_force_save_payment_method( $force_save = false, $order_id = null ) {
+		/**
+		 * Filters the flag that decides if the payment method should be saved.
+		 *
+		 * @since 9.6.0
+		 *
+		 * @param bool   $force_save Whether the payment method should be saved.
+		 * @param string $order_id   Order ID.
+		 *
+		 * @return bool Whether the payment method should be saved.
+		 */
```

Comment:
I think we need to be clearer about the behaviour. The use of "should" is probably correct, but the way it is used in IETF RFCs and other locations means that the effect of the filter could easily be misunderstood. I think we should clarify this as follows (without addressing the expectation of whether this should also apply to logged-out users):
```suggestion
		/**
		 * Filters the flag that decides if the payment method must be saved in all possible situations.
		 *
		 * @since 9.6.0
		 *
		 * @param bool   $force_save Whether the payment method must be saved.
		 * @param string $order_id   Order ID.
		 *
		 * @return bool Whether the payment method must be saved in all situations.
		 */
```